roles/rsyslog/tasks/inputs/ovirt/main.yml

```diff
@@ -9,30 +9,48 @@
   include_tasks:
     file: "{{ role_path }}/tasks/deploy.yml"
 
-- name: Create oVirt input configuration files in /etc/rsyslog.d
+- name: "Create oVirt input configuration file
+  input-ovirt-{{ _input.name }} in /etc/rsyslog.d"
   vars:
     __rsyslog_packages: []
     __rsyslog_rules:
-      - name: "input-ovirt-{{ item.name }}"
+      - name: "input-ovirt-{{ _input.name }}"
         type: input
         sections:
           - options: "{{ lookup('template', 'input_ovirt.j2') }}"
-        state: "{{ item.state | d('present') }}"
+        state: "{{ _input.state | d('present') }}"
   include_tasks:
     file: "{{ role_path }}/tasks/deploy.yml"
-  loop: '{{ logging_inputs }}'
   when:
-    - item.type | d('') == 'ovirt'
-    - item.subtype | d('') in ['collectd','engine','vdsm']
+    - _input.type | d('') == 'ovirt'
+    - _input.subtype | d('') in ['collectd','engine','vdsm']
 
 - name: Allow rsyslog to listen on collectd port
-  seport:
-    ports: "{{ item.rsyslog_read_collectd_port | d(44514) }}"
-    proto: tcp
-    setype: syslogd_port_t
-    state: present
-  loop: '{{ logging_inputs }}'
+  command: >-
+    semanage port -a -t syslogd_port_t -p tcp
+    "{{ _input.rsyslog_read_collectd_port }}"
   when:
-    - item.type | d() == 'ovirt'
+    - _input.type | d() == 'ovirt'
+    - _input.rsyslog_read_collectd_port is defined
+    - _input.rsyslog_read_collectd_port | int > 0
+    - ansible_selinux.mode in ["enforcing", "permissive"]
+  register: __result0
+  notify: restart rsyslogd
+
+- name: Check the default collectd port exists or not
+  shell: |-
+    set -euo pipefail
+    semanage port -l | grep 44514
+  ignore_errors: true
+  register: __result1
+  changed_when: false
+
+- name: "Allow rsyslog to listen on the default collectd port if no
+  other port is available and the port is not yet configured"
+  command: >-
+    semanage port -a -t syslogd_port_t -p tcp 44514
+  when:
+    - __result0.skipped
+    - __result1.rc != 0
```

Comment:
I think this sequence should be something like this (pseudocode):
```
if semanage port -l lists port {{ item.rsyslog_read_collectd_port | d(44514) }} && type syslogd_port_t && protocol tcp; then
  do nothing - already set
else
  semanage port -a -t syslogd_port_t -p tcp {{ item.rsyslog_read_collectd_port | d(44514) }} 
```
Can `semanage port -a` operate idempotently?  That is - it would be nice if `semanage port -a` would do something like this: return 0 if changed successfully; return 1 if the provided settings are already set; return -1 if there is an error - I don't see anything like that, nor does the man page give any information
Does the `seport` module reload the policy, or does it work like `semanage port --noreload`?
What does `seport` do if port `{{ item.rsyslog_read_collectd_port | d(44514) }}` is already set to some other type other than `syslogd_port_t`?  What does `semanage port -a -t syslogd_port_t -p tcp {{ item.rsyslog_read_collectd_port | d(44514) }}` do?